src/transforms/push.ts

```diff
@@ -12,20 +12,23 @@ import { JobOptions } from "bull";
 
 const logger = getLogger("transforms.push");
 
-function mapFile(githubFile) {
+const mapFile = (repoName: string, repoOwner: string, commitHash: string) => (githubFile) => {
 	// changeType enum: [ "ADDED", "COPIED", "DELETED", "MODIFIED", "MOVED", "UNKNOWN" ]
 	// on github when a file is renamed we get two "files": one added, one removed
 	const mapStatus = {
 		added: "ADDED",
 		removed: "DELETED",
 		modified: "MODIFIED"
 	};
+
+	const fallbackUrl = `https://github.com/${repoOwner}/${repoName}/blob/${commitHash}/${githubFile.filename}`
+
 	return {
 		path: githubFile.filename,
 		changeType: mapStatus[githubFile.status] || "UNKNOWN",
 		linesAdded: githubFile.additions,
 		linesRemoved: githubFile.deletions,
-		url: githubFile.blob_url
+		url: githubFile.blob_url || fallbackUrl
```

Comment:
The main part. Sometimes we are getting null here, so I decided to construct URL manually in such cases. (Even if manually constructed URL will be wrong, it will be a smaller problem, hence it won't fail the Jira request with 400, and commit files are usually not viewed)